openedx/core/djangoapps/courseware_api/urls.py

```diff
@@ -7,18 +7,23 @@
 from django.conf.urls import url
 
 from openedx.core.djangoapps.courseware_api import views
+from openedx.core.djangoapps.content.learning_sequences.data import USAGE_KEY_HASH_PATTERN
+
 
 urlpatterns = [
-    url(fr'^course/{settings.COURSE_KEY_PATTERN}',
+    url(fr'^course/{settings.COURSE_KEY_PATTERN}$',
```

Comment:
Since these are terminal URL patterns, these should have always ended with `$`.

The lack of the `$` was causing a `sequence/<usage_key_hash>` to match as `sequence/block-v1` when a usage key was provided.

Reply:
Doh! Good call. For some reason, I just assumed Django would put that on the end for us in URL parsing.